src-tauri/src/aux_functions/datatypes.rs

```diff
@@ -0,0 +1,27 @@
+use app::protobufs::{Position, User};
+
+/*
+* This is the info we're given for each node, which records (in
+* protobuf structs) its user and position info and user/position info for each neighbor.
+*
+* We'll recieve N of these and use them to create a graph with N nodes.
+*/
+pub struct NeighborInfo {
+    pub(crate) user: User,
+    pub(crate) position: Position,
+    pub(crate) num_neighbors: u32,
+    pub(crate) neighbors: Vec<Neighbor>,
```

Comment:
Why do you have pub(crate) for these? pub(crate) creates a public module so that other files can use it. Is there a module for num_neighbors or neighbors?